src/Replication/ReplicationStrategy.php

```diff
@@ -62,7 +62,8 @@ public function isReadOperation(CommandInterface $command)
         }
 
         if (($eval = $id === 'EVAL') || $id === 'EVALSHA') {
-            $sha1 = $eval ? sha1($command->getArgument(0)) : $command->getArgument(0);
+            $argument = $command->getArgument(0);
+            $sha1 = $eval ? sha1(strval($argument)) : $argument;
```

Comment:
changing `sha1(strval($argument))` to `sha1((string)$argument)` would give us a small speed